projects/observability/src/shared/dashboard/widgets/top-n/data/top-n-data-source.model.ts

```diff
@@ -46,41 +33,62 @@ export class TopNDataSourceModel extends GraphQlDataSourceModel<TopNWidgetDataFe
 
   public getData(): Observable<TopNWidgetDataFetcher> {
     return of({
-      scope: this.entityType,
-      getData: (metricSpecification: MetricAggregationSpecification) => this.fetchDataWithMetric(metricSpecification)
+      scope: this.context,
+      getData: (metricSpecification: ExploreSpecification) => this.fetchDataWithMetric(metricSpecification)
     });
   }
 
-  private fetchDataWithMetric(metricSpecification: MetricAggregationSpecification): Observable<TopNWidgetValueData[]> {
-    return this.queryWithNextBatch<EntitiesGraphQlQueryHandlerService, EntitiesResponse>(filters => ({
-      requestType: ENTITIES_GQL_REQUEST,
-      entityType: this.entityType,
+  private readonly specBuilder: ExploreSpecificationBuilder = new ExploreSpecificationBuilder();
+
+  private readonly nameAttributeSpec: ExploreSpecification = this.specBuilder.exploreSpecificationForKey('name');
```

Comment:
These values are different in different contexts - which is why they were part of the definition before. If we want to limit this data source to a subset of contexts where this holds, then this is fine but we should change up the naming + typing to indicate that.

Reply:
I am taking that up as the next step. Will add support for overriding/individually specifying name/id and context. First, I wanted everything to be converted to be explorer queries that works fine as is.

Will use appropriate names for now.

Reply:
It makes sense to make it work as is, but since this is already changing the api, I don't think it makes sense to change it once to something explorer-centric, then back to something more entity centric. Ultimately, the explorer implementation isn't super relevant to the json, so I'd push for changing the naming here. If the name/id are always the same for the supported entity types, we can leave them hardcoded.